mlir/lib/Conversion/ConvertToSPIRV/ConvertToSPIRVPass.cpp

```diff
@@ -37,20 +37,53 @@ using namespace mlir;
 namespace {
 
 /// A pass to perform the SPIR-V conversion.
-struct ConvertToSPIRVPass final
-    : impl::ConvertToSPIRVPassBase<ConvertToSPIRVPass> {
+class ConvertToSPIRVPass
+    : public impl::ConvertToSPIRVPassBase<ConvertToSPIRVPass> {
+  using impl::ConvertToSPIRVPassBase<
+      ConvertToSPIRVPass>::ConvertToSPIRVPassBase;
```

Comment:
nit: I think this can be

```suggestion
  using impl::ConvertToSPIRVPassBase::ConvertToSPIRVPassBase;
```